src/components/dataset/query/sidebar/panels/CreateSnapshotPanel.jsx

```diff
@@ -23,40 +22,60 @@ const styles = theme => ({
   cancelButton: {
     alignSelf: 'flex-end',
   },
+  textField: {
+    backgroundColor: theme.palette.common.white,
+    borderRadius: '5px',
+    marginBottom: theme.spacing(1),
+  },
 });
 
-export class CreateSnapshotPanel extends React.PureComponent {
-  static propTypes = {
-    classes: PropTypes.object,
-    handleCreateSnapshot: PropTypes.func,
-    handleSaveSnapshot: PropTypes.func,
-  };
-
-  render() {
-    const { classes, handleCreateSnapshot, handleSaveSnapshot } = this.props;
+function CreateSnapshotPanel(props) {
+  const { classes, handleCreateSnapshot, handleSaveSnapshot } = props;
+  const [name, setName] = useState('');
+  const [description, setDescription] = useState('');
 
-    return (
-      <div className={clsx(classes.rowTwo, classes.saveButtonContainer)}>
-        <TextField id="snapshotName" label="Snapshot Name" />
-        <div className={classes.buttonContainer}>
-          <Button
-            variant="contained"
-            className={clsx(classes.saveButton, { [classes.hide]: !open })}
-            onClick={() => handleCreateSnapshot(false)}
-          >
-            Cancel
-          </Button>
-          <Button
-            variant="contained"
-            className={clsx(classes.cancelButton, { [classes.hide]: !open })}
-            onClick={() => handleSaveSnapshot()}
-          >
-            Save Snapshot
-          </Button>
-        </div>
+  return (
+    <div className={clsx(classes.rowTwo, classes.saveButtonContainer)}>
+      <TextField
+        id="snapshotName"
+        label="Name"
+        variant="outlined"
+        size="small"
+        fullWidth
+        className={classes.textField}
+        onChange={event => setName(event.target.value)}
```

Comment:
Could the class-based version of this be `event => this.setState('name', event.target.value)` ?

Reply:
I tried that, and it didn't seem to work. It would only work if I passed in a function that basically wrapped `setState`. I can look into it again because I don't remember the exact error but I could have just misinterpreted it

Reply:
Ah that might have to do with how `this` works inside of functions. One workaround is to declare `const self = this;` at the top of the render function and then inside of the `onChange` callback say `self.setState(...)`.

Useful read: https://stackoverflow.com/questions/3127429/how-does-the-this-keyword-work